### PR TITLE
Support non-trivial horizontal coordinates for single column grids

### LINKED ISSUE #3626: Allow single column grids with a specific location
We want to be able to build single column grids with `topology=(Flat, Flat, Bounded)`, but with non-trivial locations (either `x, y` for `RectilinearGrid` or `latitude` and `longitude` for `LatitudeLongitudeGrid`). I think the user interface should be something like

```julia
grid = LatitudeLongitudeGrid(size=100, latitude=45, longitude=0, z=(-100, 0), topology=(Flat, Flat, Bounded))
```

This will allow us to write code that automatically interpolates from another grid (say one associated with JRA55) to the location of `grid`.

### PR DESCRIPTION
**Pull request by glwagner**
This PR makes it so we can build single column grids with non-trivial horizontal coordinates and nice syntax.

Resolves #3626 

@simone-silvestri

### CODE REVIEW

**Comment by glwagner** [comment]
The kind of thing we want to do:

```julia
from_grid = RectilinearGrid(size=(4, 4, 4), x=(1, 2), y=(2, 3), z=(0.5, 2.5), topology=(Periodic, Periodic, Bounded))
xyz = CenterField(from_grid)
set!(xyz, (x, y, z) -> x + y + z)
Oceananigans.BoundaryConditions.fill_halo_regions!(xyz)

column_grid = RectilinearGrid(size=2, x=1.3, y=2.7,  z=(0.5, 2.5), topology=(Flat, Flat, Bounded))
xyz_column = CenterField(column_grid)
Oceananigans.Fields.interpolate!(xyz_column, xyz)
```

which gives 

```julia
julia> xyz_column[1, 1, 1:2]
2-element Vector{Float64}:
 5.0
 6.0
```

**Comment by glwagner** [comment]
@simone-silvestri @navidcy please review

File: src/Fields/interpolate.jl [hunk 1]
```diff
@@ -314,11 +313,34 @@ Interpolate `field` to the physical point `(x, y, z)` using trilinear interpolat
...
 
     i, j, k = @index(Global, NTuple)
-    to_node = node(i, j, k, to_grid, to_location...)
+
+    #=
+    ℓx, ℓy, ℓz = to_location
+    ξ = ξnode(i, j, k, to_grid, ℓx, ℓy, ℓz)
+    η = ηnode(i, j, k, to_grid, ℓx, ℓy, ℓz)
+    ζ = ζnode(i, j, k, to_grid, ℓx, ℓy, ℓz)
+    =#
```
Comment (simone-silvestri) [inline src/Fields/interpolate.jl:339]: can we remove this?

**Review by simone-silvestri — APPROVED** [review]

**Comment by glwagner** [comment]
@navidcy any idea what the documenter error is?

File: src/Fields/interpolate.jl (hunk 1 above)
Comment (glwagner) [reply src/Fields/interpolate.jl]: done

**Pull request merged**